bundle/jsky.app.ot/src/main/scala/jsky/app/ot/ags/BagsManager.scala

```diff
@@ -101,25 +102,45 @@ final class BagsManager(executor: ScheduledThreadPoolExecutor) {
       synchronized {
         val key = obs.getNodeKey
         state += key
-        executor.schedule(new Thread {
-          setPriority(Thread.NORM_PRIORITY - 1)
 
-          override def run(): Unit =
+        def bagsStatus(status: BagsStatus) = bagsStatuses.synchronized {
+          val oldState = bagsStatuses.get(key)
+          bagsStatuses += ((key, status))
+          notifyBagsStatusListeners(observation, oldState, Some(status))
+        }
+        def bagsClearStatus() = bagsStatuses.synchronized {
+          val oldState = bagsStatuses.get(key)
+          bagsStatuses -= key
+          notifyBagsStatusListeners(observation, oldState, None)
+        }
+
```

Comment:
I had initially tried to put `bagsStatus` as part of `BagsState`, but was receiving a deadlock, I believe. Honestly, I didn't have much time to look into it due to rushing to complete everything, but now that you mention it, I suppose it doesn't make a lot of sense to synch on the var, does it?

I'll look into it with a bit more detail.